src/control/Control.cpp

```diff
@@ -1378,6 +1382,37 @@ void Control::changePageBackgroundColor() {
     }
 }
 
+void Control::changePageForegroundColor() {
+    int pNr = getCurrentPageNo();
+    this->doc->lock();
+    auto const& p = this->doc->getPage(pNr);
+    this->doc->unlock();
+
+    if (!p) {
+        return;
+    }
+
+    clearSelectionEndText();
+
+    PageType bg = p->getBackgroundType();
+    if (bg.isSpecial()) {
+        return;
+    }
+
+    SelectBackgroundColorDialog dlg(this);
+    dlg.show(GTK_WINDOW(this->win->getWindow()));
+
+    if (auto optColor = dlg.getSelectedColor(); optColor) {
+        PageType pageType = p->getBackgroundType();
+        auto* backgroundConfig = new BackgroundConfig(pageType.config);
+        backgroundConfig->setValueHex("f1", ((uint32_t)*optColor) - 0xff000000);
+        pageType.config = backgroundConfig->toString();
```

Comment:
Won't this leak memory? Am I missing a reason for why you can't just stack allocate it, e.g.
`BackgroundConfig backgroundConfig(pageType.config)`?

Reply:
I'm still fairly new to c++, so I'm still learning. I've changed the foreground color config setter to use stack allocation for the config parser.